### DISCUSSION TITLE
how to set it up on linux?

### DISCUSSION THREAD

**Discussion by adsick** [neovide/neovide#812] [Q&A]
Hi, I'm a not so deep into linux administration.

I've installed neovim, builded neovide from source and when I run ./target/release/neovide it works as expected.

but here is the questions: how to make neovide 'global' so I type smth like 'nvd' anywhere and it runs neovide.
also how to make it visible to my desktop environment (Pop_OS - Gnome) so I can run it using apps menu or from dock?

I know that's nooby questions, but idk how to do this things.
also is there a step by step guides on using neovim and neovide in particular (how to set up and customize it) what do you recommend?

**Comment by j4qfrost** [comment]
I can look into this, sorry I've been deep into another project lately. There should be a way to add icon installation capabilities to the snap.

**Comment by zoomlogo** [comment]
>  how to make neovide 'global' so I type smth like 'nvd' anywhere and it runs neovide

Keep neovide in `~/.local/bin/`. You should be able to invoke it with `neovide` in the terminal.

**Reply by adsick** [reply]
yep, I've already figured it out, but what if I want to run it from my desktop environment (GUI) as well?
e.g. neovim itself (installed with apt package manager) has its own icon in apps list.

**Reply by TimB87** [reply]
to make this answer complete: from the neovide source root you would want to do `install -m644 assets/neovide.desktop ~/.local/share/applications/neovide.desktop`
This will install the desktop file needed to be able to choose neovide as a text editor in menus like "open with dialogs".

Edit: make sure `~/.local/bin` is actually in your `PATH` => `echo $PATH` and if it is not, change your PATH (depends on the shell you use.. `echo $SHELL`)

**Reply by CadeMichael** [reply]
I tried this in void and it shows the neovide application with no icon and doesn't open when selected from the application menu. I built neovide in a ~/Git/packages/neovide and copied the executable to ~/.local/bin. I'm not sure what is preventing a gui app from being instantiated (It's probably an easy fix I'm just not sure if it is because I have the build in a random dir).

**Reply by TimB87** [reply]
then you didn't move its icon in place, it's equivalent to `/usr/share/icons` and probably goes into the folder `hicolor/scalable/apps/` and `hicolor/16x16/apps/` and so on..

**Reply by abdelkareemkobo** [reply]
I am using pop os and build neovim from source and it runs ok from terminal but the GUI doesn't show so i tried this command 
install -m644 assets/neovide.desktop ~/.local/share/applications/neovide.desktop
but get cannot stat 'asset/neovide.desktop': no such file or directory 
any help?

**Reply by ReiCat** [reply]
You can just copy `neovide.desktop` file to the `~/.local/share/applications/` and it should work. If you want to add an icon just copy it too and add `Icon=~/.local/share/applications/{icon-name.png}` variable in `neovide.desktop`.
